arrow/src/main/java/org/apache/calcite/adapter/arrow/ArrowFieldType.java

```diff
@@ -25,27 +25,17 @@
 
 import java.math.BigDecimal;
 import java.sql.Date;
-import java.util.List;
 
 import static java.util.Objects.requireNonNull;
 
 /**
  * Arrow field type.
  */
-enum ArrowFieldType {
-  INT(Primitive.INT),
-  BOOLEAN(Primitive.BOOLEAN),
-  STRING(String.class),
-  FLOAT(Primitive.FLOAT),
-  DOUBLE(Primitive.DOUBLE),
-  DATE(Date.class),
-  LIST(List.class),
-  DECIMAL(BigDecimal.class),
-  LONG(Primitive.LONG),
-  BYTE(Primitive.BYTE),
-  SHORT(Primitive.SHORT);
+class ArrowFieldType {
```

Comment:
I think enums are preferable, since they are typechecked by the compiler.
A Class can be anything.

Reply:
Later, I may dynamically pass some parameters according to the data type of arrow, which is difficult to achieve through enumeration.

Reply:
Why later if you are refactoring the code now?
Do it correctly the first time.
I am suggesting to have 3 arguments for the enum constructor: Class, precision, scale.

Reply:
@mihaibudiu Sorry for the late reply, you are right. I looked at the arrow types and it seems that Class, precision, and scale are sufficient.

Reply:
I am not sure I understand the question.
There are lots of enum classes with parameters in the codebase.
See https://github.com/apache/calcite/blob/main/linq4j/src/main/java/org/apache/calcite/linq4j/tree/Primitive.java for example

Reply:
@mihaibudiu thanks, done

Reply:
The strange thing is that when I don't use an enumeration to pass in the precision, the CLI will not report an error

Reply:
What is the CLI?

Reply:
> What is the CLI?

Sorry. it is CI

Reply:
I'm using an enum class to represent this, but it doesn't seem ideal, CI always fails.
https://github.com/apache/calcite/actions/runs/12014265742/job/33489690898

Reply:
Have you tried to apply the recommendations made in the output of the failed CI run?
```
/home/runner/work/calcite/calcite/core/src/main/java/org/apache/calcite/sql2rel/RelDecorrelator.java:2654: Note: [MethodCanBeStatic] A private method that does not reference the enclosing instance can be static
    private void onMatch2(
    ^
    (see https://errorprone.info/bugpattern/MethodCanBeStatic)
  Did you mean 'private static void onMatch2('?
/home/runner/work/calcite/calcite/core/src/main/java/org/apache/calcite/util/format/postgresql/format/compiled/NumberCompiledPattern.java:82: Note: [MethodCanBeStatic] A private method that does not reference the enclosing instance can be static
  private String getOrdinalSuffix(String stringValue) {
  ^
    (see https://errorprone.info/bugpattern/MethodCanBeStatic)
  Did you mean 'private static String getOrdinalSuffix(String stringValue) {'?
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :arrow:compileJava
/home/runner/work/calcite/calcite/arrow/src/main/java/org/apache/calcite/adapter/arrow/ArrowTranslator.java:233: Note: [MethodCanBeStatic] A private method that does not reference the enclosing instance can be static
  private String translateUnaryOp(String op, String name) {
  ^
    (see https://errorprone.info/bugpattern/MethodCanBeStatic)
  Did you mean 'private static String translateUnaryOp(String op, String name) {'?
/home/runner/work/calcite/calcite/arrow/src/main/java/org/apache/calcite/adapter/arrow/ArrowFieldType.java:47: warning: [ImmutableEnumChecker] enums should be immutable: 'ArrowFieldType' has non-final field 'precision'
  private int precision;
              ^
    (see https://errorprone.info/bugpattern/ImmutableEnumChecker)
  Did you mean 'private final int precision;'?
/home/runner/work/calcite/calcite/arrow/src/main/java/org/apache/calcite/adapter/arrow/ArrowFieldType.java:48: warning: [ImmutableEnumChecker] enums should be immutable: 'ArrowFieldType' has non-final field 'scale'
  private int scale;
              ^
    (see https://errorprone.info/bugpattern/ImmutableEnumChecker)
  Did you mean 'private final int scale;'?
```